### PR TITLE
HDDS-9128. Ozone freon tool extension for benchmarking listKeys operation

### PR DESCRIPTION
**Pull request by tanvipenumudy**
## What changes were proposed in this pull request?

Enhance the read/write tool (ockrw) to include list operation support. Users can specify the percentage of read, write, and list tasks for generating the mixed workload. This should help in benchmarking the performance of the listKeys operation.
 
The below command runs 50% of read-key tasks, 40% list-key tasks and 10% of write-key tasks from pre-generated keys:
`ozone freon ockrw --percentage-read 50 --percentage-list 40 -r 100 -v voltest -b buckettest -p performanceTest`

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9128

## How was this patch tested?

- Added robot tests.
- Green CI/CD: https://github.com/tanvipenumudy/ozone/actions/runs/5787028346/job/15698237642
- The updated freon tool has been tested over a cluster for capturing the metric: `om_performance_metrics_list_keys_latency_ns_avg_time` (in nanoseconds) - PR: #5131

listKeys_latency

### CODE REVIEW

**Comment by tanvipenumudy** [comment]
@DaveTeng0 could you please take a look at this as well? Thanks.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyReadWriteListOps.java [hunk 1]
```diff
@@ -212,16 +230,22 @@ keyName, objectSizeInBytes, null, new HashMap())) {
...
     }
-    //mix workload
+  }
+
+  public TaskType decideReadWriteOrListTask() {
     int tmp = ThreadLocalRandom.current().nextInt(1, 101);
-    if (tmp  <= percentageRead) {
+    if (tmp <= percentageRead) {
       return TaskType.READ_TASK;
+    } else if (percentageRead < tmp && tmp <= percentageRead + percentageList) {
```
Comment (kerneltime) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyReadWriteListOps.java:247]:
```suggestion
    } else if (tmp > percentageRead && tmp <= percentageRead + percentageList) {
```

**Review by kerneltime — APPROVED** [review]

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyReadWriteListOps.java [hunk 2]
```diff
@@ -212,16 +230,22 @@ keyName, objectSizeInBytes, null, new HashMap())) {
...
     }
-    //mix workload
+  }
+
+  public TaskType decideReadWriteOrListTask() {
     int tmp = ThreadLocalRandom.current().nextInt(1, 101);
-    if (tmp  <= percentageRead) {
+    if (tmp <= percentageRead) {
       return TaskType.READ_TASK;
+    } else if (tmp > percentageRead && tmp <= percentageRead + percentageList) {
```
Comment (duongkame) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyReadWriteListOps.java:247]:
```suggestion
    } else if (tmp <= percentageRead + percentageList) {
```
redundant condition `tmp > percentageRead` as it's true already.

**Review by duongkame — APPROVED** [review]
LGTM, +1. 
Added a super minor suggestion.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyReadWriteListOps.java (hunk 2 above)
Comment (DaveTeng0) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyReadWriteListOps.java:247]: good catch!

**Comment by DaveTeng0** [comment]
> @DaveTeng0 could you please take a look at this as well? Thanks.

Thanks for the changes! It looks good to me, except a minor suggestion from Duong~

**Pull request merged**